tests/jenkins/TestOpenSearchReleasePromotionParallelTest.groovy

```diff
@@ -0,0 +1,168 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import jenkins.tests.BuildPipelineTest
+import org.junit.Before
+import org.junit.Test
+import org.yaml.snakeyaml.Yaml
+import static com.lesfurets.jenkins.unit.global.lib.LibraryConfiguration.library
+import static com.lesfurets.jenkins.unit.global.lib.GitSource.gitSource
+import static com.lesfurets.jenkins.unit.MethodCall.callArgsToString
+import static org.hamcrest.CoreMatchers.hasItem
+import static org.hamcrest.MatcherAssert.assertThat
+import static org.junit.jupiter.api.Assertions.assertThrows
+
+class TestOpenSearchReleasePromotionTestParallel extends BuildPipelineTest {
+
+    @Override
+    @Before
+    void setUp() {
+
+        helper.registerSharedLibrary(
+            library().name('jenkins')
+                .defaultVersion('5.11.1')
+                .allowOverride(true)
+                .implicit(true)
+                .targetPath('vars')
+                .retriever(gitSource('https://github.com/opensearch-project/opensearch-build-libraries.git'))
+                .build()
+            )
+
+        super.setUp()
+        addParam('RELEASE_VERSION', '1.0.0')
+        addParam('OPENSEARCH_RC_BUILD_NUMBER', '2050')
+        addParam('OPENSEARCH_DASHBOARDS_RC_BUILD_NUMBER', '3050')
+        binding.setVariable('AGENT_LINUX_X64', 'Jenkins-Agent-AL2023-X64-C54xlarge-Docker-Host')
+    }
+
+    @Test
+    void shouldExecuteWithoutErrors() {
```

Comment:
Is there a way to also check that job was actually executed in parallel? Since assertions only state it got executed.

Reply:
Not really! What the end state of the job would be can only be mocked from our end or run actually on the jenkins.

Reply:
hmm, I wish there was a mechanism to latch or clock counter.